src/test/java/components/task/DeadlineTaskTest.java

```diff
@@ -0,0 +1,55 @@
+package components.task;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+
+import utilities.DateTime;
+
+public class DeadlineTaskTest {
+
+    private DeadlineTask buildDeadline(String description, boolean done, LocalDateTime deadline) {
+        String[] tokens = new String[] { TaskType.DEADLINE.toString(), description, done ? "1" : "0",
+                DateTime.formatDateTime(deadline) };
+        return (DeadlineTask) DeadlineTask.decodeData(tokens);
+    }
+
+    @Test
+    void encodeDecode_roundTrip() {
+        LocalDateTime dl = LocalDateTime.now().plusDays(2).withSecond(0).withNano(0);
+        DeadlineTask task = buildDeadline("Submit report", false, dl);
+        String encoded = task.encodeData();
+        String[] tokens = encoded.split(utilities.Data.DELIMITER);
+        DeadlineTask decoded = (DeadlineTask) DeadlineTask.decodeData(tokens);
+        assertEquals("Submit report", decoded.getDescription());
+        assertFalse(decoded.isDone());
+    }
+
+    @Test
+    void isDueSoon_trueWithin7DaysAndNotDone() {
```

Comment:
Perhaps you could include an additional underscore to denote the expected behavior instead of it being in the test scenario field.
```suggestion
    void isDueSoon_dueInLessThan7DaysAndNotDone_returnsTrue() {
```